pkg/reconciler/taskrun/taskrun.go

```diff
@@ -433,6 +433,19 @@ func (c *Reconciler) prepare(ctx context.Context, tr *v1beta1.TaskRun) (*v1beta1
 		return nil, nil, controller.NewPermanentError(err)
 	}
 
+	if tr.Status.TaskRunResults == nil || len(tr.Status.TaskRunResults) == 0 {
+		tr.Status.TaskRunResults = make([]v1beta1.TaskRunResult, 0)
+		for _, tR := range rtr.TaskSpec.Results {
+			if tR.Default != nil {
+				tr.Status.TaskRunResults = append(tr.Status.TaskRunResults, v1beta1.TaskRunResult{
+					Name:  tR.Name,
+					Type:  tR.Type,
+					Value: *tR.Default,
+				})
+			}
```

Comment:
I think this part needs to be tested. Seems to be missing in the coverage report.

Reply:
ie.the test case needs to trigger`if tR.Default != nil {`